lib/server/controllers/file_controller.rb

```diff
@@ -104,4 +107,91 @@ def copy
     )
     success_json(files: [ new_file.to_hash(@request) ])
   end
+
+  def bulk_copy
+    # We want to support bulk copy operations
+    #   from Magma.
+    # First, check that the user has access to all of
+    #   the revision buckets and files, and that
+    #   all revisions are valid.
+    # If all revisions are valid, execute them.
+    require_param(:revisions)
+
+    revisions = @params[:revisions]
+
+    raise Etna::BadRequest, 'At least one revision required' unless revisions.length > 0
+
+    raise Etna::BadRequest, 'All revisions require "source" parameter' unless revisions.all? {|revision| revision.has_key? :source}
+    raise Etna::BadRequest, 'All revisions require "dest" parameter' unless revisions.all? {|revision| revision.has_key? :dest}
+
+    all_source_buckets = revisions.select {|rev| rev[:source]}.
+      map {|rev| extract_bucket_from_path(rev[:source])}.uniq
+    all_dest_buckets = revisions.select {|rev| rev[:dest]}.
+      map {|rev| extract_bucket_from_path(rev[:dest])}.uniq
+
```

Comment:
This might be nice in a revision object that collects all of these validations and bucket-extractions and so on.